### PR TITLE
Feature/venn diagrams

### PR DESCRIPTION
**Pull request by phpfs**
This PR introduces a VennDiagram component capable of rendering venn diagrams of 4th order (or lower).

### CODE REVIEW

File: app/src/components/VennDiagram/venn/types/types.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import * as d3 from 'd3';
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/venn/types/types.ts:1]: Import only necessary things and not all

File: app/src/components/VennDiagram/VennDiagramProps.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+export interface VennDiagramProps {
+  flavor: VennDiagramFlavors;
+  payload: VennTwoSetsPayload | VennThreeSetsPayload | VennFourSetsPayload;
+}
+
+export enum VennDiagramFlavors {
+  TwoSets,
+  ThreeSets,
+  FourSets,
+}
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/VennDiagramProps.ts:14]: It should be a separate file

File: app/src/components/VennDiagram/VennDiagram.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+import { VennDiagramTooltip } from 'components/VennDiagram/venn/tooltip';
+import { stageTwoCirclesOn } from 'components/VennDiagram/venn/twoSets';
+import { VennFourSetsPayload } from 'components/VennDiagram/venn/types/fourSetsTypes';
+import { VennThreeSetsPayload } from 'components/VennDiagram/venn/types/threeSetsTypes';
+import { VennTwoSetsPayload } from 'components/VennDiagram/venn/types/twoSetsTypes';
+import {
+  VennDiagramFlavors,
+  VennDiagramProps,
+} from 'components/VennDiagram/VennDiagramProps';
+import * as d3 from 'd3';
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/VennDiagram.tsx:12]: Import only necessary things

File: app/src/components/VennDiagram/VennDiagram.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+import React, { useEffect } from 'react';
+
+export const VennDiagram = ({
+  flavor,
+  payload,
+}: VennDiagramProps): JSX.Element => {
+  // Attention - will be updated for prop update!
+  let svgElement: SVGSVGElement | null = null;
+  let tooltip: VennDiagramTooltip | undefined = undefined;
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/VennDiagram.tsx:21]: Do not use let within a functional component

File: app/src/components/VennDiagram/venn/fundamentals/draw.ts [hunk 5]
```diff
@@ -0,0 +1,163 @@
+/*------------ See README.txt for license and copyright information! -----------*/
+import { VennDiagramTooltip } from 'components/VennDiagram/venn/tooltip';
+import { d3Selection } from 'components/VennDiagram/venn/types/types';
+import * as d3 from 'd3';
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/venn/fundamentals/draw.ts:4]: Import only necessary things

File: app/src/components/VennDiagram/venn/tooltip.ts [hunk 6]
```diff
@@ -0,0 +1,44 @@
+/*------------ See README.txt for license and copyright information! -----------*/
+import { drawTooltip } from 'components/VennDiagram/venn/fundamentals/draw';
+import { d3Selection } from 'components/VennDiagram/venn/types/types';
+
+export interface VennDiagramTooltipProps {
+  element: d3Selection;
+  id?: string;
+}
```
Comment (florian-papsdorf) [inline app/src/components/VennDiagram/venn/tooltip.ts:8]: It should be a separate file

**Review by florian-papsdorf — CHANGES_REQUESTED** [review]
Overall really good job, think of some improvements that streamline changes related to our existing work and do not use any let's within a functional component

File: app/src/components/VennDiagram/VennDiagramProps.ts (hunk 2 above)
Comment (martingraf4) [reply app/src/components/VennDiagram/VennDiagramProps.ts]: has been removed

File: app/src/components/VennDiagram/VennDiagram.tsx (hunk 4 above)
Comment (martingraf4) [reply app/src/components/VennDiagram/VennDiagram.tsx]: uses useRef and useState now

File: app/src/components/VennDiagram/venn/tooltip.ts (hunk 6 above)
Comment (martingraf4) [reply app/src/components/VennDiagram/venn/tooltip.ts]: has been removed

**Review by martingraf4 — APPROVED** [review]

**Review by phpfs — COMMENTED** [review]
Nice improvements :)

**Pull request merged**